pkg/cmd/admin/router/router.go

```diff
@@ -438,7 +440,8 @@ func RunCmdRouter(f *clientcmd.Factory, cmd *cobra.Command, out io.Writer, cfg *
 			return fmt.Errorf("router could not be created; you must specify a service account with --service-account")
```

Comment:
Yeah, we should move this check for service account out and if possible fix it, so that we only do the checks when its relevant. Plus we should do all the validations before we get into the generate part.